compiler/rustc_typeck/src/check/method/probe.rs

```diff
@@ -1246,24 +1247,46 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
         &self,
         stable_pick: &Pick<'_>,
         unstable_candidates: &[(&Candidate<'tcx>, Symbol)],
+        self_ty: Ty<'tcx>,
     ) {
         self.tcx.struct_span_lint_hir(
             lint::builtin::UNSTABLE_NAME_COLLISIONS,
             self.fcx.body_id,
             self.span,
             |lint| {
-                let mut diag = lint.build(
-                    "a method with this name may be added to the standard library in the future",
-                );
+                let def_kind = stable_pick.item.kind.as_def_kind();
+                let mut diag = lint.build(&format!(
+                    "{} {} with this name may be added to the standard library in the future",
+                    def_kind.article(),
+                    def_kind.descr(stable_pick.item.def_id),
+                ));
                 // FIXME: This should be a `span_suggestion` instead of `help`
                 // However `self.span` only
                 // highlights the method name, so we can't use it. Also consider reusing the code from
                 // `report_method_error()`.
```

Comment:
I guess this FIXME is now half-resolved? can you update it or move it into the first match arm?

Reply:
My interpretation of the FIXME is that we still have the same problem for method calls. We are only handling the assoc const case correctly now (which we didn't handle at all before). I can move it to the appropriate arm though.